### DISCUSSION TITLE
Bug: the use of always events in parallel transition requests causes "Maximum call stack size exceeded".

### DISCUSSION THREAD

**Discussion by fronterior** [statelyai/xstate#3503] [Q&A] [answered]
### Description

[Example](https://codesandbox.io/s/xstate-always-maximum-call-stack-size-exceeded-yxbj3e?file=/src/index.ts)

I'm trying to separate routing and logic through status. The screen is in loading state during loading of common resources, and the page can be viewed after loading. The example machine was created briefly, but the loaded resource also includes the state of the stored screen, and the function to switch to the saved screen state after loading is also used. So I decided that there was a state concerning resource loading and a state regarding screen at the same time, and I used the parallel state.

In an example, when the resource is loaded, it causes a transition to the screen state. I prefer to be automatically transitioned to the conditions on the machine if possible, rather than sending an event. Because I think it's more natural in expressing logic. So I wanted to send always event that causes transition from the final state to the parallel state. However, this creates an exception `Maximum call stack size exceeded`. It seems that always events occur several times in a row, is this intended behavior? In contrast, the after event works well.

### Expected result

always events should occur only once and no exceptions `Maximum call stack size exceeded` should occur.

### Actual result

After events work well, but always events cause exceptions `Maximum call stack size exceeded`.

### Reproduction

https://codesandbox.io/s/xstate-always-maximum-call-stack-size-exceeded-yxbj3e?file=/src/index.ts

### Additional context

I am using XState in the Node.js environment. So I use `@xstate/inspect/lib/server`
- @xstate/inspect: 0.7.0
- xstate: 4.32.1

It's also being reproduced in codesandbox. You can check by sending events `ALWAYS_TEST` and `AFTER_TEST` from the inspector.

**Comment by davidkpiano** [comment, marked as answer]
There is a workaround for your use-case: **multiple targets**

```js
on: {
  ALWAYS_TEST: {
    target: ["alwaysDone", "#alwaysBugMachine.screen.login"]
  },
},

// ...

alwaysDone: {
  type: "final"
  // always: {
  //   target: "#alwaysBugMachine.screen.login"
  // }
},
```

https://user-images.githubusercontent.com/1093738/183294697-53ecdcc8-9b76-4087-b2f9-bce1908d4fcb.mp4

**Reply by fronterior** [reply]
Thank. Since the multiple target function is not yet available in the editor, it is unfortunate that the multiple target will be erased if I modify it in the editor, but it works well. Is this intended to cause the exception? Or can it be modified later?

**Reply by davidkpiano** [reply]
There's probably another way to model this without having to use multiple targets, such as specifying another transition for `ALWAYS_TEST` in the `screen` region.

The editor will eventually support multiple targets.
